dashboard/pages/1_Insights.py

```diff
@@ -88,62 +91,133 @@ def show_recent_average_and_median(connection):
     median_df = pd.read_sql(LAST_HOUR_MEDIAN_SCORE, connection)
     return average_df["avg"].round().iloc[0], median_df["median_score"].iloc[0]
 
-    # grouped_df = df.groupby(['story_id', 'record_time']).sum().reset_index()
-
-    # st.write("How comments change over time")
-
-    # return st.line_chart(grouped_df.set_index('record_time'))
-
 
 def show_top_authors(connection):
-    
+    """Produces bar chart of the top authors in terms of contributions
+    and score of stories
+    """
+
     df1 = pd.read_sql(MOST_AUTHOR_CONTRIBUTIONS, connection)
     df2 = pd.read_sql(MOST_POPULAR_AUTHORS, connection)
-    # Display the results in Streamlit
     st.title("Top Authors Overview")
 
-    # Display the results of the first SQL query
-    # st.header("Authors with Most Contributions")
-    # st.table(df1)
-
-    # # Display the results of the second SQL query
-    # st.header("Authors with Highest Scores")
-    # st.table(df2)
-
-    # Plot bar charts for better visualization
     fig, ax = plt.subplots(2, 1, figsize=(10, 12))
 
-    # Bar chart for stories created
     ax[0].bar(df1['author'], df1['stories_created'], color='skyblue')
     ax[0].set_title('Top Authors by Stories Created')
     ax[0].set_ylabel('Number of Stories Created')
     ax[0].set_xlabel('Authors')
     ax[0].grid(axis='y', linestyle='--', alpha=0.7)
 
-    # Bar chart for total votes
     ax[1].bar(df2['author'], df2['all_stories_total_votes'], color='lightcoral')
     ax[1].set_title('Top Authors by Total Votes')
     ax[1].set_ylabel('Total Votes')
     ax[1].set_xlabel('Authors')
     ax[1].grid(axis='y', linestyle='--', alpha=0.7)
 
-    # Adjust layout to prevent overlap
     plt.tight_layout()
 
-    # Display the plots in Streamlit
     st.pyplot(fig)
 
+
+def show_five_biggest_movers(connection):
+    """Produces line chart of 5 biggest movers in the last 24 hours
+    """
+
+    df_records = pd.read_sql(FIVE_BIGGEST_MOVERS, connection)
+    df_records['record_time'] = pd.to_datetime(df_records['record_time'])
+
+    fig, ax = plt.subplots(figsize=(10, 6))
+
+    for title in df_records['title'].unique():
+        data_for_story = df_records[df_records['title'] == title]
+        ax.plot(data_for_story['record_time'], data_for_story['score'], label=f'{title}')
+
+    # Set labels and title
+    ax.set_xlabel('Timestamp')
+    ax.set_ylabel('Votes')
+    ax.set_title('5 biggest movers over last 24 hours')
+    ax.legend()
+
+    ax.set_xticklabels(df_records['record_time'], rotation=45, ha='right')
+    ax.xaxis.set_major_formatter(mdates.DateFormatter('%d/%m %H:%M'))
+
+    st.pyplot(fig)
+
+
+def extract_publisher(url):
+    """Extracts the publisher from a given story url
+    """
+    pattern = re.compile(r"https?://(?:www\.)?(.*?)\/(.*)")
+
+    match = pattern.search(url)
+
+    if match:
+        result = match.group(1)
+        if result.count(".") == 2:
+            result = result.split(".")
+            return result[1]
```

Comment:
Might as well return `result.split(".")[1]` in one line because in the next line you're doing that (`result.split(".")[0]`)